src/orchestratord/src/controller/materialize/resources.rs

```diff
@@ -1143,6 +1161,239 @@ fn create_statefulset_object(
     }
 }
 
+fn create_balancerd_deployment_object(config: &super::Args, mz: &Materialize) -> Deployment {
+    let limits = btreemap! {
+        "cpu".to_string() =>
+            Quantity(mz.balancerd_cpu_allocation(&config.default_balancerd_cpu_allocation)),
+        "memory".to_string() =>
+            Quantity(mz.balancerd_memory_allocation(&config.default_balancerd_memory_allocation)),
+    };
+    let requests = btreemap! {
+        "cpu".to_string() =>
+            Quantity(mz.balancerd_cpu_allocation(&config.default_balancerd_cpu_allocation)),
+        "memory".to_string() =>
+            Quantity(mz.balancerd_memory_allocation(&config.default_balancerd_memory_allocation)),
+    };
+
+    let security_context = if config.enable_security_context {
+        // Since we want to adhere to the most restrictive security context, all
+        // of these fields have to be set how they are.
+        // See https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted
+        Some(SecurityContext {
+            run_as_non_root: Some(true),
+            capabilities: Some(Capabilities {
+                drop: Some(vec!["ALL".to_string()]),
+                ..Default::default()
+            }),
+            seccomp_profile: Some(SeccompProfile {
+                type_: "RuntimeDefault".to_string(),
+                ..Default::default()
+            }),
+            allow_privilege_escalation: Some(false),
+            ..Default::default()
+        })
+    } else {
+        None
+    };
+
+    let mut pod_template_labels = mz.default_labels();
+    pod_template_labels.insert(
+        "materialize.cloud/name".to_owned(),
+        mz.balancerd_deployment_name(),
+    );
+    pod_template_labels.insert("app".to_owned(), "balancerd".to_string());
+
+    let ports = vec![ContainerPort {
+        container_port: config.balancerd_internal_http_port,
+        name: Some("internal-http".into()),
+        protocol: Some("TCP".into()),
+        ..Default::default()
+    }];
+
+    let args = vec![
+        "service".to_string(),
+        format!("--pgwire-listen-addr=0.0.0.0:{}", config.balancerd_sql_port),
+        format!("--https-listen-addr=0.0.0.0:{}", config.balancerd_http_port),
+        format!(
+            "--internal-http-listen-addr=0.0.0.0:{}",
+            config.balancerd_internal_http_port
+        ),
+        format!(
+            "--https-resolver-template={}.{}.svc.cluster.local:{}",
+            mz.environmentd_service_name(),
+            mz.namespace(),
+            config.environmentd_internal_balancer_port
+        ),
+        format!(
+            "--static-resolver-addr={}.{}.svc.cluster.local:{}",
+            mz.environmentd_service_name(),
+            mz.namespace(),
+            config.environmentd_sql_port
```

Comment:
Are we able to do this just because we don't have TLS, and the balancers currently don't use TLS to talk to environmentd?

If so, why not do the same with the HTTP port?

Reply:
yeah, i'll make this consistent